### DISCUSSION TITLE
TLS/SSL: No Shared Cipher

### DISCUSSION THREAD

**Discussion by sireskay** [postalserver/postal#1987] [Help with using Postal] [answered]
I have looked at every discussion relating to TLS/SSL and did not find the topic about this specific issue.
I am using Sectigo assigned Certificate, and when I reviewed the log from testing the Server, I got the following: 

**SSL Negotiation Failed: SSL_accept returned=1 errno=0 state=error: no shared cipher**

_### What I have tried_

- copying the ciphers from the initiating server into tls_ciphers, assigned 
- using Let-encrypt cert (that was a different error)
- assigned several ssl_version

Any help or direction will be appreciated

**Comment by willpower232** [comment]
You can verify that your Postal server is offering a certificate with `openssl s_client -connect postal.yourdomain.com:25 -starttls smtp`

If it is, then that means that whatever is trying to send that message has an out of date certificate bundle.

**Reply by sireskay** [reply]
Thanks, @willpower232!

**Comment by sireskay** [comment, marked as answer]
Here is what I did to get it working if someone stumbles here for a solution.

I had to use the buddle certificate from Sectigo.

convert p7b to PEM
openssl pkcs7 -in Certificate.p7b -print_certs -out certificate.pem

also converted Key from .txt to PEM
cat cetificate_key.txt > certificate_key.pem

I placed the certificated in the Config folder, referenced the file, and TLS worked.

I am now able to use port 587 and TLS.

**Reply by willpower232** [reply]
Ah yes good to know, not every certificate provider makes PEM formats easily available.
